TLS distribution: bind erts socket to localhost

legoscia opened the pull request:
There is no reason for the socket on the erts side of the proxy to
accept connections from other hosts, so let's bind it to 127.0.0.1.

This will probably conflict with some of the other pull requests related to TLS distribution; let me know if you want me to rebase it.

OTP-Maintainer commented:
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

IngelaAndin commented:
We agree with the concept, but think that we ought to use {ip, loopback} as done in odbc instead of hardcoding IPV4.  However it does  look like this feature has not been documented. And it might be so that the tls_dist is getting this wrong in other places (if I remember correctly).  Also there has been a few patches in this area that have been accepted for 18.2. So we propose that we all revisit this PR when 18.2 is out.

IngelaAndin commented:
ping

legoscia commented:
I updated the patch to use `{ip, loopback}` instead, and changed the existing occurrence of `{ip, {127,0,0,1}}`. I left the `gen_tcp:connect({127,0,0,1})` calls as they are — I couldn't think of anything obvious to do about them, and it would be easy to do something clean as part of #878.

IngelaAndin commented:
We have accepted this PR to the test phase, and stalled PR 878 that has already been accepted for the test phase, but not reached it yet. It is almost Christmas now, so we will get back to you after the holidays and together we can figure out what will be the best solution.

OTP-Maintainer commented:
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

IngelaAndin commented:
We agreed that localhost should be the way to solve connect for now. We have crated a backlog item for documenting {ip, loopback} and creating a loopback alternative for connect.  But we will accept #878 with localhost.
